src/components/pages/Transaction/Transaction.tsx

```diff
@@ -1,12 +1,24 @@
+import { useEffect } from 'react';
 import { Navigate, useLocation } from 'react-router-dom';
 
 import { useTransactionData } from '../../../hooks';
-import TransactionWorkflow from '../../layout/TransactionWorkflow/TransactionWorkflow';
+import { useTransactionState } from '../../../state/contexts/TransactionState';
+import TransactionWorkflow, {
+  TRANSACTION_WORKFLOW_STATUS,
+} from '../../layout/TransactionWorkflow/TransactionWorkflow';
 
 function Transaction() {
   const { transactionData, contractType, interactionType, workflowStage } =
     useTransactionData();
   const from = useLocation().state;
+  const [{}, dispatchTransactionState] = useTransactionState(); // eslint-disable-line
+
+  useEffect(() => {
+    dispatchTransactionState({
+      type: 'setWorkflowStage',
+      payload: TRANSACTION_WORKFLOW_STATUS.PENDING,
+    });
```

Comment:
what about `reset` at the end of `TransactionWorkflow` instead of force updating here?

Reply:
if we reset the workflow stage it will rerender us back to the pending stage..... that needs to be done either at the beginning or in the state provider.... even in the state provider im not sure that works, i think it has to always be at the beginning. eg when the transaction page loads.

Reply:
hmm, this might cause issues when we do add in dynamic URL loading - as it'll ignore what `useTransactionData` hook will parse and set in the provider state

Reply:
for now, let's do this. we'll confirm that the page was loaded from an internal page when setting this. that way, any internal routing will start fresh, and when we add dynamic linking, `from` will be empty, so it won't override the state set by `useTransactionData`

```
  useEffect(() => {
    if (from) {
      dispatchTransactionState({
        type: 'setWorkflowStage',
        payload: TRANSACTION_WORKFLOW_STATUS.PENDING,
      });
    }
  }, [from]);
```

Reply:
fine as is for now, just wanted to call this out.